extension-style/src/main/java/com/mapbox/maps/extension/style/sources/generated/GeoJsonSource.kt

```diff
@@ -594,27 +597,47 @@ class GeoJsonSource(builder: Builder) : Source(builder.sourceId) {
   ): GeoJsonSource = apply {
     onDataParsed?.let { listener ->
       ignoreParsedGeoJsonRegistry[sourceId] = false
-      // remove any events from queue before posting this task
-      workerHandler.removeCallbacksAndMessages(null)
-      workerHandler.post {
-        val property = data.toPropertyValue()
-        mainHandler.post {
-          // we set parsed data when sync setter was not called during background work
-          if (ignoreParsedGeoJsonRegistry[sourceId] == false) {
-            setProperty(property, throwRuntimeException = false)
-            listener.invoke(this)
+      // No parseJob is working now, parse the data directly.
```

Comment:
In other places `workerHandler.removeCallbacksAndMessages(null)` was replace by `waitingData = null` but in this case some another logic appears.

Reply:
HandlerThread has a queue and could pop the events automatically, we don't have the queue and need to handle the logic mannually.

Reply:
my high level q is more about that this manual logic is used only here while in code above is technically almost the same and we do not perform similar logic there.
but I did not give it a deep look - I'll check out your branch later to validate it myself but please focus on retaining logic 1 to 1 as it was before, we need to be careful not to bring in unexpected bugs 😉

Reply:
`code above` here means the part in the constructor?  One instance can't be constructed twice, so the logic for that part can be simpler than this part as this function can be invoked multiple times